hera_cal/abscal_funcs.py

```diff
@@ -1059,201 +1216,125 @@ def __eq__(self, B2):
-
-    # load models
-    for i, mf in enumerate(model_files):
-        echo("loading {}".format(mf), type=0, verbose=verbose)
-        uv = UVData()
-        try:
-            uv.read_miriad(mf)
-        except:
-            uv.read_uvfits(mf)
-            uv.unphase_to_drift()
-        if i == 0:
-            uvm = uv
+    # create baseline keys for model
+    model_keys = model.keys()
+    model_bls = np.array(map(lambda k: Baseline(model_antpos[k[1]] - model_antpos[k[0]], tol=tol), model_keys))
+
+    # create baseline keys for data
+    data_keys = data.keys()
+    data_bls = np.array(map(lambda k: Baseline(data_antpos[k[1]] - data_antpos[k[0]], tol=tol), data_keys))
+
+    # iterate over data baselines
+    new_model = odict()
+    for i, bl in enumerate(model_bls):
+        # compre bl to all model_bls
+        comparison = np.array(map(lambda mbl: bl == mbl, data_bls), np.str)
+
+        # get matches
+        matches = np.where((comparison=='True')|(comparison=='conjugated'))[0]
+
+        # check for matches
+        if len(matches) == 0:
+            echo("found zero matches in data for model {}".format(model_keys[i]), verbose=verbose)
+            continue
         else:
-            uvm += uv
-
-    model, mflags, mpols = UVData2AbsCalDict([uvm], pol_select=pols)
-    for i, k in enumerate(model.keys()):
-        if k[0] == k[1]:
-            model.pop(k)
-
-    # get model params
-    model_times = np.unique(uvm.lst_array)
-    model_freqs = uvm.freq_array.squeeze()
+            if len(matches) > 1:
+                echo("found more than 1 match in data to model {}: {}".format(model_keys[i], map(lambda j: data_keys[j], matches)), verbose=verbose)
+            # assign to new_data
+            if comparison[matches[0]] == 'True':
+                new_model[data_keys[matches[0]]] = model[model_keys[i]]
+            elif comparison[matches[0]] == 'conjugated':
+                new_model[data_keys[matches[0]]] = np.conj(model[model_keys[i]])
 
-    # align model freq-time axes to data axes
-    model = interp_model(model, model_times, model_freqs, data_times, data_freqs,
-                        kind='cubic', fill_value=0, zero_tol=1e-6)
+    return DataContainer(new_model)
 
-    # check if model has only unique baseline data
-    # this is the case if, for example, the model Nbls in less than the data Nbls
-    if uvm.Nbls < uvd.Nbls:
-        # try to expand model data into redundant baseline groups
-        model = mirror_data_to_red_bls(model, bls, antpos, tol=2.0)
 
-        # ensure data keys match model keys
-        for i, k in enumerate(data):
-            if k not in model:
-                data.pop(k)
+def avg_data_across_red_bls(data, antpos, flags=None, broadcast_flags=True, median=False, tol=1.0,
+                            mirror_red_data=False):
+    """
+    Given complex visibility data spanning one or more redundant
+    baseline groups, average redundant visibilities and write to file
```

Comment:
I think this docstring is out of date. You don't write to file here.

Reply:
yup